pkg/asset/cluster/tfvars.go

```diff
@@ -270,6 +272,34 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 		if err != nil {
 			return err
 		}
+
+		baseImage := string(*rhcosImage)
+		var baseImageURL string
+
+		// If base image name is <InfraID>-rhcos then it was not provided by the user and we have to create it.
+		// Since Glance image creation is not idempotent (there can be multiple images with the same name),
+		// we need to find a way to explain Terraform when to create an image and when not. For doing this
+		// we get the image URL and give it to Terraform as a variable. Therefore, if openstack-base-image-url
+		// is empty, then Terraform will reuse the existing image, otherwise create a new one from the given
+		// url. Image name in this case will be: <InfraID>-rhcos, for instance, "clustername-rdd6e-rhcos".
+		if baseImage == clusterID.InfraID+"-rhcos" {
+			var err error
+			ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
+			defer cancel()
```

Comment:
What is this doing?  It looks like it's always cancelling a context that is unused?